index.js

```diff
@@ -479,14 +487,14 @@ module.exports = class Hypercore extends EventEmitter {
     if (this.replicator !== null) {
       this.replicator.findingPeers -= this._findingPeers
       this.replicator.clearRequests(this.activeRequests, err)
-      this.replicator.updateActivity(this._active ? -1 : 0, this)
+      this.replicator.updateActivity(this._active ? -1 : 0)
     }
 
     this._findingPeers = 0
 
-    if (this.sessions.length) {
+    if (this.sessions.length || this.core.active > 0) {
```

Comment:
Sanity check on the intent of the flow:
It's possible for `this.core.active`to be > 0, while `this.sessions.length` is 0, right? And in that case we can just close without worrying about closing any sessions (since there are none).

Reply:
correct, active is the "correct" number always